server/project/project.go

```diff
@@ -229,6 +255,16 @@ func (s *Server) Get(ctx context.Context, q *project.ProjectQuery) (*v1alpha1.Ap
 		}
 		proj.Spec.Roles[i] = role
 	}
+
+	for role, roleTokenEntry := range proj.Status.JWTTokenMap {
```

Comment:
The project normalization is not only token ID normalization. We need to make sure that both `proj.Status.JWTTokenMap` and `proj.Status.Roles.JWTTokens` has the same tokens. 

So we need to copy missing tokens from `proj.Status.Roles.JWTTokens` to `proj.Status.JWTTokenMap` and the other way around. 

I suggest creating method `Normalize` ( `func (p *AppProject) Normalize()`  ) in `types.go` the normalize tokens IDs and perform copying.  Then execute `Normalize` in every method that updates/returns project.

Reply:
Thank you Alex for your detailed review! Normalization is implemented https://github.com/argoproj/argo-cd/pull/3791/files#diff-0956fb34f840d2ef6b93253499eaeee4R440